src/main/java/pl/javastart/task/logic/MixPhoneContract.java

```diff
@@ -0,0 +1,83 @@
+package pl.javastart.task.logic;
+
+public class MixPhoneContract extends Contract {
+    private static final int SMS_LIMIT = 100;
+    private static final int MMS_LIMIT = 50;
+    private static final int SECONDS_LIMIT = 60;
+    private static final int SECONDS_IN_MINUTE = 60;
+    private double accountBalance;
+    private int sumPhoneCallSeconds;
+
+    public MixPhoneContract(double smsCost, double mmsCost, double oneMinuteCallCost, double accountBalance) {
+        super(smsCost, oneMinuteCallCost, mmsCost);
+        this.accountBalance = accountBalance;
+    }
+
+    public double getAccountBalance() {
+        return accountBalance;
+    }
+
+    private void setAccountBalance(double accountBalance) {
+        this.accountBalance = accountBalance;
+    }
+
+    @Override
+    public void sendSms() {
+        if (sentSms <= SMS_LIMIT) {
+            smsSent();
+            sumSentSms();
+        } else if (getAccountBalance() >= getSmsCost()) {
+            System.out.println("SMS wysłany, kosztował: " + getSmsCost() + " zł");
+            sumSentSms();
+            accountBalance -= getSmsCost();
+        } else {
+            noMoneyToSendSms();
+        }
```

Comment:
Jeżeli w tej klasie dziedziczyłbyś po `CardPhoneContract`, to ten fragment można zastąpić wywołaniem `super.sendSms()`